File: src/main/java/hse/btf/pdfeditor/models/Item.java
package hse.btf.pdfeditor.models;

// внешний прямоугольник, в который обрамлен контент
public abstract class Item {
    // default width and height
    private static final int defaultWidth = 150;
    private static final int defaultHeight = 80;

    // field center
    private static final int centerX = 388;
    private static final int centerY = 272;
    /**
     * сделать размер центра изменяемым
     * подумать над минимальным размером Item'а
     * и обработать ситуации, когда размер, например, не может быть уменьшен,
     * потому что контент внутри не позволяет этого сделать
     **/

    private int x = centerX;
    private int y = centerY;
    private int w = defaultWidth;
    private int h = defaultHeight;
    private String color = "white";
}

File: src/main/kotlin/hse/btf/pdfeditor/models/Item.kt
package hse.btf.pdfeditor.models

import hse.btf.pdfeditor.serializers.DoublePropertySerializer
import hse.btf.pdfeditor.serializers.IntegerPropertySerializer
import javafx.beans.property.DoubleProperty
import javafx.beans.property.IntegerProperty
import javafx.beans.property.SimpleDoubleProperty
import javafx.beans.property.SimpleIntegerProperty
import kotlinx.serialization.Serializable

@Serializable
sealed class Item {
    @Serializable(with = DoublePropertySerializer::class)
    val x: DoubleProperty = SimpleDoubleProperty(150.0)

    @Serializable(with = DoublePropertySerializer::class)
    val y: DoubleProperty = SimpleDoubleProperty(200.0)

    @Serializable(with = IntegerPropertySerializer::class)
    val w: IntegerProperty = SimpleIntegerProperty(100)

    @Serializable(with = IntegerPropertySerializer::class)
    val h: IntegerProperty = SimpleIntegerProperty(50)
}